src/test/java/com/atlassian/db/replica/it/example/aurora/AuroraClusterTest.java

```diff
@@ -63,21 +61,18 @@ void shouldUtilizeReplicaForReadQueriesForSynchronisedWrites() throws SQLExcepti
             readerJdbcUrl,
             writerJdbcUrl
         );
-        final DatabaseCluster cluster = new AuroraCluster(
-            connectionProvider::getMainConnection,
-            readerEndpoint,
-            databaseName
-        );
         final ReplicaConsistency replicaConsistency = new ConsistencyFactory(
             connectionProvider::getMainConnection,
             connectionProvider,
-            cluster
+            readerEndpoint,   // TODO: I added the params here, but I'm sure databaseName can be discovered, not sure about readerEndpoint.
```

Comment:
> I added the params here, but I'm sure databaseName can be discovered, not sure about readerEndpoint.

It would require an additional call to db. I would lean more towards a solution where we don't need to pass connection for discovery. I think it serves as optimization, but once we implement an internal pool it will become obsolete.

Reply:
> It would require an additional call to db.

It's just one call. It can be cached as it doesn't change.

> I would lean more towards a solution where we don't need to pass connection for discovery. 

Can you elaborate? How can we discover cluster topology without a connection?